### PR TITLE
End-to-end Nexmark benchmark

### PR DESCRIPTION
**Pull request by blp**
This is a first stab at an end-to-end benchmark test.

Is this a user-visible change (yes/no): no

### CODE REVIEW

File: demo/nexmark/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Nexmark
+
+These instructions will allow you to run the Nexmark benchmark using
```
Comment (mihaibudiu) [inline demo/nexmark/README.md:3]: The queries are written in SQL.
Currently only a few of the queries from the benchmark are executed.
Notice that there is a separate Nexmark benchmark where more queries are implemented in Rust.

File: demo/nexmark/nexmark.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
+CREATE TABLE person (
+    id BIGINT,
+    name VARCHAR,
+    emailAddress VARCHAR,
+    creditCard VARCHAR,
+    city VARCHAR,
+    state VARCHAR,
+    date_time TIMESTAMP(3), -- NOT NULL LATENESS INTERVAL 4 SECONDS,
```
Comment (mihaibudiu) [inline demo/nexmark/nexmark.sql:8]: now we support lateness and watermarks for these columns, which could potentially affect performance significantly

File: demo/nexmark/nexmark.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+FROM (
+   SELECT A.*, B.auction, B.bidder, B.price, B.date_time AS bid_dateTime, B.extra AS bid_extra,
+     ROW_NUMBER() OVER (PARTITION BY A.id ORDER BY B.price DESC, B.date_time ASC) AS rownum
+   FROM auction A, bid B
+   WHERE A.id = B.auction AND B.date_time BETWEEN A.date_time AND A.expires
+)
+WHERE rownum <= 1;
+
+-- -------------------------------------------------------------------------------------------------
+-- Query 10: Log to File System (Not in original suite)
```
Comment (mihaibudiu) [inline demo/nexmark/nexmark.sql:51]: clearly, this doesn't involve a filesystem... 
so perhaps some of these comments could be modified

File: demo/nexmark/run.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    pipeline_name = "nexmark-pipeline"
+    requests.put(f"{api_url}/v0/pipelines/{pipeline_name}", json={
+        "description": "",
+        "config": {"workers": 6},
+        "program_name": program_name,
+        "connectors": connectors,
+    }).raise_for_status()
+
+    # Stop pipeline
+    print("Stopping pipeline...")
```
Comment (mihaibudiu) [inline demo/nexmark/run.py:100]: why is this necessary?
I would make each of these into a little function
they will probably be in a library soon...

**Review by mihaibudiu — APPROVED** [review]
any other instructions on how to get more perf data except just completion time?

File: demo/nexmark/nexmark.sql [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    date_time TIMESTAMP(3), -- NOT NULL LATENESS INTERVAL 4 SECONDS,
+    extra  VARCHAR
+);
+-- -------------------------------------------------------------------------------------------------
+-- Query 9: Winning Bids (Not in original suite)
+-- -------------------------------------------------------------------------------------------------
+-- Find the winning bid for each auction.
+-- -------------------------------------------------------------------------------------------------
+
+CREATE VIEW q9 AS
```
Comment (lalithsuresh) [inline demo/nexmark/nexmark.sql:38]: We'd be executing both queries at once in this benchmark. Maybe leave a note saying they should be split later?

**Review by lalithsuresh — APPROVED** [review]

File: demo/nexmark/nexmark.sql (hunk 5 above)
Comment (ryzhyk) [inline demo/nexmark/nexmark.sql:38]: nexmark is supposed to be run one query at a time. Do you plan to support that too, so we can compare the results against Flink and others?

File: demo/nexmark/nexmark.sql (hunk 2 above)
Comment (blp) [reply demo/nexmark/nexmark.sql:8]: Do I just uncomment these?

Comment (mihaibudiu) [reply demo/nexmark/nexmark.sql:8]: you can, but the semantics is slightly different.
for example, they will skip processing tuples that are too much out of order
so this may make the pipeline very fast

File: demo/nexmark/nexmark.sql (hunk 3 above)
Comment (blp) [reply demo/nexmark/nexmark.sql:51]: This comment comes from your NexmarkTest.java.

Comment (mihaibudiu) [reply demo/nexmark/nexmark.sql:51]: It was copied by Michael from the Java flink code

File: demo/nexmark/nexmark.sql (hunk 5 above)
Comment (gz) [reply demo/nexmark/nexmark.sql:38]: fwiw being faster than X and running all queries at once might also be a powerful statement.. if we can

Comment (blp) [reply demo/nexmark/nexmark.sql]: The new version I just pushed runs one query at a time, or you can tell it `--merge` and it will run the requested queries together.

Comment (blp) [reply demo/nexmark/nexmark.sql]: The new version I just pushed runs one query at a time, or you can tell it --merge and it will run the requested queries together.

I ran a few queries together and they took about the same time as running just one of them individually.

File: demo/nexmark/run.py (hunk 4 above)
Comment (blp) [reply demo/nexmark/run.py]: I added functions.

File: demo/nexmark/nexmark.sql (hunk 3 above)
Comment (blp) [reply demo/nexmark/nexmark.sql]: I deleted all the SQL comments. They aren't helpful here.

File: demo/nexmark/nexmark.sql (hunk 2 above)
Comment (blp) [reply demo/nexmark/nexmark.sql]: The version I just pushed uses lateness by default, or you can specify `--no-lateness` to turn if off.
